x/gov/abci.go

```diff
@@ -46,6 +46,12 @@ func EndBlocker(ctx sdk.Context, keeper *keeper.Keeper) error {
 			return false, err
 		}
 
+		// HV2: heimdall always refunds and deletes deposits in all cases of proposal failures, without caring about params.BurnProposalDepositPrevote
+		err = keeper.RefundAndDeleteDeposits(ctx, proposal.Id) // refund deposit if proposal got removed without getting 100% of the proposal
```

Comment:
Concern here is potential for spam...the Hub had a pretty significant spam proposal issue so you might want to consider some of the previous arguments for alternative solutions

Reply:
Ok, thanks, let's explore this in the call today, then we will evaluate internally.

Reply:
[Added](39fd8a993160d716f503d10c9f5567f397c4605f) a `TODO` in the code to keep track of it, will discuss internally.

Reply:
Agree. Will take this up with gov team (burn, FeeCollector deposit, or any alternative).
Made a note in our internal migration doc.
Resolving the comment.